packages/wasm/src/address.ts

```diff
@@ -1,14 +1,21 @@
 import { get_short_address_by_index, is_controlled_address } from '../wasm';
-import { AddressIndex } from '@buf/penumbra-zone_penumbra.bufbuild_es/penumbra/core/keys/v1/keys_pb';
+import {
+  Address,
+  AddressIndex,
+} from '@buf/penumbra-zone_penumbra.bufbuild_es/penumbra/core/keys/v1/keys_pb';
 import { JsonValue } from '@bufbuild/protobuf';
+import { bech32Address } from '@penumbra-zone/bech32/src/address';
 
 export const getShortAddressByIndex = (fullViewingKey: string, index: number) =>
   get_short_address_by_index(fullViewingKey, index) as string;
 
-export const isControlledAddress = (
+export const getAddressIndexByAddress = (
   fullViewingKey: string,
-  address: string,
+  address: Address,
 ): AddressIndex | undefined => {
-  const res = is_controlled_address(fullViewingKey, address) as JsonValue;
+  const res = is_controlled_address(fullViewingKey, bech32Address(address)) as JsonValue;
```

Comment:
Think we should consider changing the rust api as well. 

`is_controlled_address()` should return a boolean

`get_address_index_by_address()` should return Result<AddressIndex> (throws if not able to)